README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# <img src="man/figures/hex-med.png" width="25%" align="right"/> youngSwimmers

<!-- badges: start -->

[![R-CMD-check](https://github.com/NIM-ACh/youngSwimmers/workflows/R-CMD-check/badge.svg)](https://github.com/NIM-ACh/youngSwimmers/actions)
[![Lifecycle:
stable](https://img.shields.io/badge/lifecycle-stable-brightgreen.svg)](https://lifecycle.r-lib.org/articles/stages.html#stable)
[![CRAN
status](https://www.r-pkg.org/badges/version/youngSwimmers)](https://CRAN.R-project.org/package=youngSwimmers)
<!-- badges: end -->

This is an `R` package that contains the data used for the young
swimmers [study](https://doi.org/10.3389/fphys.2021.769085), which aims
to be a complementary tool for research reproducibility and exploratory
data analysis.

## About <i class="fab fa-r-project"></i>

[`R`](https://www.r-project.org/about.html) is a high-level statistical
programming language used for data analysis, as well for development
creating software, websites, reports, applications, e-books, etc.

## Installation

You can install the released version of `youngSwimmers` from
[CRAN](https://CRAN.R-project.org/package=youngSwimmers) with:

``` r
install.packages("youngSwimmers")
```

To install the latest development version from
[GitHub](https://github.com/) use:

``` r
# install.packages("devtools")
devtools::install_github("nim-ach/youngSwimmers")
```

## Example

This is a basic example which shows you how to access the dataset:

``` r
library(youngSwimmers) # Load the package

swimmers[, 1:6] # Print columns 1 to 6
#>      period subject    sex age weight height
#>   1:    t-0     s01   Male  13   66.0  166.0
#>   2:    t-0     s02   Male  16     NA  176.0
#>   3:    t-0     s03 Female  15   62.8  158.0
#>   4:    t-0     s05 Female  12   45.6  157.5
#>   5:    t-0     s06   Male  15   63.5  168.0
#>  ---                                        
#> 126:    t-4     s34 Female  14   62.4  158.5
#> 127:    t-4     s36   Male  12   51.5  164.0
#> 128:    t-4     s40 Female  13   50.0  159.0
#> 129:    t-4     s45   Male  13   50.9  164.0
#> 130:    t-4     s46 Female  14   69.3  159.0
```

For more examples on how to use the dataset, please visit the section
[“Using the `swimmers`
dataset”](https://nim-ach.github.io/youngSwimmers/articles/articles/using_data.html)
from our website. There you will find information for computing
descriptive statistics and plotting functionalities.

## Citation

To cite the dataset please use:

Castillo-Aguilar M, Valdés-Badilla P, Herrera-Valenzuela T, Guzmán-Muñoz
E, Delgado-Floody P, Andrade DC, Moraes MM, Arantes RME and
Núñez-Espinosa C. (2021). Cardiac Autonomic Modulation in Response to
Muscle Fatigue and Sex Differences During Consecutive Competition
Periods in Young Swimmers: A Longitudinal Study. Front. Physiol.
12:769085. doi: 10.3389/fphys.2021.769085

Or run from your `R` console:

``` r
citation("youngSwimmers")
#> 
#> To cite youngSwimmers in publications use:
#> 
#>   Castillo-Aguilar M, Valdés-Badilla P, Herrera-Valenzuela T,
#>   Guzmán-Muñoz E, Delgado-Floody P, Andrade DC, Moraes MM, Arantes RME
#>   and Núñez-Espinosa C. (2021). Cardiac Autonomic Modulation in
#>   Response to Muscle Fatigue and Sex Differences During Consecutive
#>   Competition Periods in Young Swimmers: A Longitudinal Study. Front.
#>   Physiol. 12:769085. doi: 10.3389/fphys.2021.769085
#> 
#> A BibTeX entry for LaTeX users is
#> 
#>   @Article{,
#>     title = {Cardiac Autonomic Modulation in Response to Muscle Fatigue and Sex Differences During Consecutive Competition Periods in Young Swimmers: A Longitudinal Study},
#>     author = {Matías Castillo-Aguilar and Pablo Valdés-Badilla and Tomás Herrera-Valenzuela and Eduardo Guzmán-Muñoz and Pedro Delgado-Floody and David Cristóbal Andrade and Michele M. Moraes and Rosa M. E. Arantes and Cristian Núñez-Espinosa},
#>     journal = {Frontiers in Physiology},
#>     year = {2021},
#>     volume = {12},
#>     pages = {1989},
#>     url = {https://www.frontiersin.org/articles/10.3389/fphys.2021.769085/full},
#>     doi = {10.3389/fphys.2021.769085},
#>     issn = {1664-042X},
#>   }
```


NEWS.md
# youngSwimmers 0.0.1

* Added a `NEWS.md` file to track changes to the package.


R/data.R
#' Data from the young elite swimmers study
#'
#' This is the data used for the young elite swimmers
#' study (Castillo-Aguilar et al., 2021). It contains records from
#' 26 competitive swimmers from ages 10 to 16 on 5 different
#' competitive time periods.
#'
#' @format This is a data.table object containing 27 variables and 130 rows
#'
#' - \code{period}: Factor. Time periods from two competitions.
#' - \code{subject}: Factor. Subject ID.
#' - \code{sex}: Factor. Subject's sex (Male of Female).
#' - \code{age}: Numeric. Subject's age in years.
#' - \code{weight}: Numeric. Weight in kilograms.
#' - \code{height}: Numeric. Heigh in centimeters.
#' - \code{fat}: Numeric. Body fat in percentage.
#' - \code{bmi}: Numeric. Body mass index.
#' - \code{ffmi}: Numeric. Fat free mass index.
#' - \code{sp}: Numeric. Systolic blood pressure in mmHg.
#' - \code{dp}: Numeric. Diastolic blood pressure in mmHg.
#' - \code{map}: Numeric. Mean arterial pressure in mmHg.
#' - \code{pp}: Numeric. Pulse pressure in mmHg.
#' - \code{sdnn_pre}: Numeric. SDNN (Time domain parameter) pre-wingate test.
#' - \code{rmssd_pre}: Numeric. RMSSD (Time domain parameter) pre-wingate test.
#' - \code{vlf_pre}: Numeric. VLF (Frequency domain parameter) pre-wingate test.
#' - \code{lf_pre}: Numeric. LF (Frequency domain parameter) pre-wingate test.
#' - \code{hf_pre}: Numeric. HF (Frequency domain parameter) pre-wingate test.
#' - \code{sdnn_post}: Numeric. SDNN (Time domain parameter) post-wingate test.
#' - \code{rmssd_post}: Numeric. RMSSD (Time domain parameter) post-wingate test.
#' - \code{vlf_post}: Numeric. VLF (Frequency domain parameter) post-wingate test.
#' - \code{lf_post}: Numeric. LF (Frequency domain parameter) post-wingate test.
#' - \code{hf_post}: Numeric. HF (Frequency domain parameter) post-wingate test.
#' - \code{power_peak}: Numeric. Peak power output in Watts.
#' - \code{power_mean}: Numeric. Mean power output in Watts.
#' - \code{power_min}: Numeric. Minimum power output in Watts.
#' - \code{fatigue}: Numeric. Fatigue index in percentage.
#'
#' @source
#' <https://doi.org/10.3389/fphys.2021.769085>
"youngswimmers"


logo/making_logo.R

if (requireNamespace("hexSticker")) {
  library(hexSticker)

  img_url <- "logo/swimmer.png"

  sticker(subplot = img_url,
          package = "youngSwimmers",
          filename = "man/figures/hex-big.png",
          s_x = 1, s_y = 0.85,
          s_width = 0.5,
          p_size = 54,
          p_y = 1.45,
          h_size = 2,
          h_fill = "#69aadb", h_color = "#96c3e5",
          dpi = 1080)

  sticker(subplot = img_url,
          package = "youngSwimmers",
          filename = "man/figures/hex-med.png",
          s_x = 1, s_y = 0.85,
          s_width = 0.5,
          p_size = 30,
          p_y = 1.45,
          h_size = 2,
          h_fill = "#69aadb", h_color = "#96c3e5",
          dpi = 600)

  sticker(subplot = img_url,
          package = "youngSwimmers",
          filename = "man/figures/hex-small.png",
          s_x = 1, s_y = 0.85,
          s_width = 0.5,
          p_size = 16,
          p_y = 1.45,
          h_size = 2,
          h_fill = "#69aadb", h_color = "#96c3e5",
          dpi = 300)
} else {
  stop("Need to install hexSticker first!")
}


R/dataset.R
#' Data from the young elite swimmers study
#'
#' This is the data used for the young elite swimmers
#' study (Castillo-Aguilar et al. 2021). It contains records from
#' 26 competitive swimmers from ages 10 to 16 on 5 different
#' competitive time periods.
#'
#' @format This is a data.table object containing 27 variables and 130 rows
#'
#' - \code{period}: Factor. Time periods from two competitions.
#' - \code{subject}: Factor. Subject ID.
#' - \code{sex}: Factor. Subject's sex (Male of Female).
#' - \code{age}: Numeric. Subject's age in years.
#' - \code{weight}: Numeric. Weight in kilograms.
#' - \code{height}: Numeric. Heigh in centimeters.
#' - \code{fat}: Numeric. Body fat in percentage.
#' - \code{bmi}: Numeric. Body mass index.
#' - \code{ffmi}: Numeric. Fat free mass index.
#' - \code{sp}: Numeric. Systolic blood pressure in mmHg.
#' - \code{dp}: Numeric. Diastolic blood pressure in mmHg.
#' - \code{map}: Numeric. Mean arterial pressure in mmHg.
#' - \code{pp}: Numeric. Pulse pressure in mmHg.
#' - \code{sdnn_pre}: Numeric. SDNN (Time domain parameter) pre-wingate test.
#' - \code{rmssd_pre}: Numeric. RMSSD (Time domain parameter) pre-wingate test.
#' - \code{vlf_pre}: Numeric. VLF (Frequency domain parameter) pre-wingate test.
#' - \code{lf_pre}: Numeric. LF (Frequency domain parameter) pre-wingate test.
#' - \code{hf_pre}: Numeric. HF (Frequency domain parameter) pre-wingate test.
#' - \code{sdnn_post}: Numeric. SDNN (Time domain parameter) post-wingate test.
#' - \code{rmssd_post}: Numeric. RMSSD (Time domain parameter) post-wingate test.
#' - \code{vlf_post}: Numeric. VLF (Frequency domain parameter) post-wingate test.
#' - \code{lf_post}: Numeric. LF (Frequency domain parameter) post-wingate test.
#' - \code{hf_post}: Numeric. HF (Frequency domain parameter) post-wingate test.
#' - \code{power_peak}: Numeric. Peak power output in Watts.
#' - \code{power_mean}: Numeric. Mean power output in Watts.
#' - \code{power_min}: Numeric. Minimum power output in Watts.
#' - \code{fatigue}: Numeric. Fatigue index in percentage.
#'
#' @source
#' \doi{10.3389/fphys.2021.769085}
"swimmers"
